sycl/plugins/unified_runtime/ur/adapters/hip/enqueue.cpp

```diff
@@ -1328,20 +1383,88 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueUSMPrefetch(
 #endif
 }
 
+/// USM: memadvise API to govern behavior of automatic migration mechanisms
 UR_APIEXPORT ur_result_t UR_APICALL
 urEnqueueUSMAdvise(ur_queue_handle_t hQueue, const void *pMem, size_t size,
-                   ur_usm_advice_flags_t, ur_event_handle_t *phEvent) {
+                   ur_usm_advice_flags_t advice, ur_event_handle_t *phEvent) {
 #if HIP_VERSION_MAJOR >= 5
+  UR_ASSERT(pMem && size > 0, UR_RESULT_ERROR_INVALID_VALUE);
   void *HIPDevicePtr = const_cast<void *>(pMem);
+
+  // Passing MEM_ADVISE_SET/MEM_ADVISE_CLEAR_PREFERRED_LOCATION and
+  // to hipMemAdvise on a GPU device requires the GPU device to report a
+  // non-zero value for hipDeviceAttributeConcurrentManagedAccess. Therfore,
+  // ignore memory advise if concurrent managed memory access is not available.
+  if ((advice & UR_USM_ADVICE_FLAG_SET_PREFERRED_LOCATION) ||
+      (advice & UR_USM_ADVICE_FLAG_CLEAR_PREFERRED_LOCATION) ||
+      (advice & UR_USM_ADVICE_FLAG_SET_ACCESSED_BY_DEVICE) ||
+      (advice & UR_USM_ADVICE_FLAG_CLEAR_ACCESSED_BY_DEVICE) ||
+      (advice & UR_USM_ADVICE_FLAG_DEFAULT)) {
+    ur_device_handle_t Device = hQueue->getContext()->getDevice();
+    if (!getAttribute(Device, hipDeviceAttributeConcurrentManagedAccess)) {
+      setErrorMessage("mem_advise ignored as device does not support "
+                      "concurrent managed access",
+                      UR_RESULT_SUCCESS);
+      return UR_RESULT_ERROR_ADAPTER_SPECIFIC;
+    }
+
+    // TODO: If pMem points to valid system-allocated pageable memory, we should
```

Comment:
Is this coming as a followup?

Reply:
Most likely as a follow-up, yes. I am still trying out ways to go on about checking this stuff.